app/models/entry.rb

```diff
@@ -53,11 +53,13 @@ def calculate(type, *args)
 
       private
       def all(*args)
-        find_many :find, :all, *args
+        ActiveSupport::Deprecation.warn('Passing arguments is deprecated') if args.any?
+        find_many :all # *args
```

Comment:
From staring at the implementation I gathered this much:
The Entry class was created as a way of introducing a superclass for TimeEntry and CostEntry where there has been no superclass before (as it is not possible to change the superclass after it has been defined). Depending on the way you want to look at it, it thus has up to two concerns:
1) Be an abstract superclass for TimeEntry and CostEntry. That is where the Delegator class is used for.
2) Ensure, that the child classes TimeEntry and CostEntry respond to everything the superclass defines (e.g. `cost_type`, `units`).

I believe it would be relatively easy to separate the two concerns. That could remove the necessity of using `method_missing`, one of the levels of indirection. 

Instead of the deprecation warning, I'd simply remove the `*args` from the signature. It might fail, but if it does, it will fail hard and noticeably. I fear that we might be hunting for weird bugs soon otherwise.

Not that it would reduce indirection but the two methods `all` and `count` could also be defined like
```
%w[all count].each do |meth|
        define_method(meth) { |*args| send_all(meth) }
      end
```